core/src/main/java/com/amplitude/core/platform/plugins/ContextPlugin.kt

```diff
@@ -40,6 +40,11 @@ class ContextPlugin : Plugin {
                 event.plan = it
             }
         }
+        event.ingestionMetadata ?: let {
+            amplitude.configuration.ingestionMetadata ?. let {
```

Comment:
clone here?

Reply:
Thank you @justin-fiedler for reviewing! I create a `clone` method and clone the object here, PTAL whether this is what we want here.

I think timeline enriches the event in sequence, so the clone only prevents concurrent context plugin and enrichment plugin running case? Please let me know if I am wrong.

Reply:
Has the same question as Marvin here.
The plugins will be processed in the order of `Before` -> 'Enrichment' -> `Destination`. Feels we cannot prevent the enrichment modification.

Reply:
I am concerned about an Enrichment plugin modifying an individual Event's `plan` or `ingestionMetadata` values, which by reference would modify the shared values in `amplitude.configuration` and affect other subsequent events.

Reply:
@liuyang1520 let me know if this makes sense. If so you can mark as resolved and merge.